platform/commonUI/edit/src/actions/SaveInProgressDialog.js

```diff
@@ -1,21 +1,30 @@
 define([], function () {
-    function SaveInProgressDialog(dialogService) {
-        this.dialogService = dialogService;
-        this.dialog = undefined;
+    var self = this;
+
+    function SaveInProgressDialog(dialogService, $timeout) {
+        self.dialogService = dialogService;
+        self.$timeout = $timeout;
+
+        self.dialog = undefined;
+        self.timeoutId = undefined;
     }
 
     SaveInProgressDialog.prototype.show = function () {
```

Comment:
Let's move the declaration of the `self` variable to this function and limit its scope.